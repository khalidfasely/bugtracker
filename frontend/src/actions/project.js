import delBugApi from "../api/delBug";
import newBugApi from "../api/newBug";
import projectApi from "../api/project";

export const setProject = (project, bugs) => ({
    type: "SET_PROJECT",
    project,
    bugs
});

export const startSetProject = (pid) => {
    return (dispatch) => {
        return projectApi(pid).then((result) => {
            dispatch(setProject(result.project, result.bugs));
            return result;
        });
    };
};

export const setNewBug = (bug) => ({
    type: 'NEW_BUG',
    bug
});

export const startSetNewBug = (on_project, bugData) => {
    return (dispatch) => {
        return newBugApi(on_project, bugData).then(result => {
            if (result.bug) {
                dispatch(setNewBug(result.bug));
            }
            return result;
        });
    };
};

export const setDeleteBug = (bid) => ({
    type: 'DELETE_BUG',
    bid
});

export const startSetDeleteBug = (bid) => {
    return (dispatch) => {
        return delBugApi(bid).then(result => {
            dispatch(setDeleteBug(bid));
            return result;
        });
    };
};