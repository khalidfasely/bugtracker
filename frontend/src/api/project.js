export default async (pid) => {
    const res = await fetch(`/api/project/${pid}`);

    const project = await res.json();

    return project;
};