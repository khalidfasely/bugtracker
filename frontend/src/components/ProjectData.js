const ProjectData = ({ projectItem }) => (
    <div>
        <p>{projectItem?.id}</p>
        <p>{projectItem?.name}</p>
        <p>{projectItem?.time}</p>
        <p>{projectItem?.user?.username}</p>
        <div>
            {projectItem?.admins?.map(admin => <span key={admin}>{admin} </span>)}
        </div>
        <br />
        <div>
            {projectItem?.users_with?.map(user => <span key={user}>{user} </span>)}
        </div>
    </div>
);

export default ProjectData;