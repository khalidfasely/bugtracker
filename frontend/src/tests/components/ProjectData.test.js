import React from "react";
import { render } from "@testing-library/react";

import ProjectData from "../../components/ProjectData";
import project from "../fixtures/project";

test('Should render ProjectData component without props', () => {
    const { asFragment } = render(<ProjectData />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should render ProjectData component with props', () => {
    const { asFragment } = render(<ProjectData projectItem={project} />);
    expect(asFragment()).toMatchSnapshot();
});