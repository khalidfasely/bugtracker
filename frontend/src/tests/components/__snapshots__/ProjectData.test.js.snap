// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ProjectData component with props 1`] = `
<DocumentFragment>
  <div
    class="project-data"
  >
    <p
      class="project__name"
    >
      Project one
    </p>
    <p
      class="project__user"
    >
      <span
        class="static-text"
      >
        By: 
      </span>
      1
    </p>
    <p
      class="project__time"
    >
      <span
        class="static-text"
      >
        On: 
      </span>
      now
    </p>
    <div
      class="project__users"
    >
      <span
        class="static-text"
      >
        Users in: 
      </span>
      <span>
        1 
      </span>
    </div>
    <div
      class="project__admins"
    >
      <span
        class="static-text"
      >
        Admins in: 
      </span>
      <span>
        1 
      </span>
      <span>
        admin 
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should render ProjectData component without props 1`] = `
<DocumentFragment>
  <div
    class="project-data"
  >
    <p
      class="project__name"
    />
    <p
      class="project__user"
    >
      <span
        class="static-text"
      >
        By: 
      </span>
    </p>
    <p
      class="project__time"
    >
      <span
        class="static-text"
      >
        On: 
      </span>
    </p>
    <div
      class="project__users"
    >
      <span
        class="static-text"
      >
        Users in: 
      </span>
    </div>
    <div
      class="project__admins"
    >
      <span
        class="static-text"
      >
        Admins in: 
      </span>
    </div>
  </div>
</DocumentFragment>
`;
